### PR TITLE
En 1866 fix content list cache key

### PR DESCRIPTION
**Pull request by eugeniosant**
The problem was found in the content list widget.
when in a content list we create a filter directly on the jsp, and this filter is dynamic (for example based on the attributes of the current user), the key with which the object is cached is the same and the resulting list is always the same.
This PR fixes this problem

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java [hunk 1]
```diff
@@ -252,23 +254,18 @@ protected boolean extractOrCategoryFilterClause(ApsProperties config) {
...
         String widgetFilters = widgetParams.getProperty(widgetParamName);
         EntitySearchFilter[] filters = this.getFilters(bean.getContentType(), widgetFilters, reqCtx);
         if (null == filters) {
-            return;
-        }
-        for (EntitySearchFilter filter : filters) {
-            bean.addFilter(filter);
+            return bean.getFilters();
         }
+        EntitySearchFilter[] filtersToReturn = (null != bean.getFilters()) ? Arrays.copyOf(bean.getFilters(), bean.getFilters().length) : new EntitySearchFilter[0];
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java]: Way too complex for a trinary operator. Tough to see what is happening here. Please pul l this apart.

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java [hunk 2]
```diff
@@ -252,23 +254,18 @@ protected boolean extractOrCategoryFilterClause(ApsProperties config) {
...
-    @Deprecated
-    protected void addShowletFilters(IContentListTagBean bean, ApsProperties showletParams, String showletParamName, RequestContext reqCtx) {
-        this.addWidgetFilters(bean, showletParams, showletParamName, reqCtx);
-    }
-
-    protected void addWidgetFilters(IContentListTagBean bean, ApsProperties widgetParams, String widgetParamName, RequestContext reqCtx) {
+    protected EntitySearchFilter[] createWidgetFilters(IContentListTagBean bean, ApsProperties widgetParams, String widgetParamName, RequestContext reqCtx) {
         if (null == widgetParams) {
-            return;
+            return bean.getFilters();
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java]: This looks like a pretty big change possibly. Given that nothing would get set in the old call. Is this covered by a test and do we know that the contract is correct if `widgetParams` is null?

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java (hunk 1 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java]: Removed trinary operator

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java (hunk 2 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java]: The signature of the method has changed; before the modification the method added the filters into the "bean" object (erroneously changing the key with which the list was searched in cache), now it creates and returns a list of filters without altering the "bean" object.
BTW, I added other junit tests in order to cover other cases.

**Pull request merged**